src/jsifier.js

```diff
@@ -302,6 +290,9 @@ function JSify(data, functionsOnly) {
       if ((EXPORT_ALL || (finalName in EXPORTED_FUNCTIONS)) && !noExport) {
         contentText += '\nModule["' + finalName + '"] = ' + finalName + ';';
       }
+      if (MAIN_MODULE && sig) {
```

Comment:
maybe a comment here as to why we need this `.sig` property? not obvious to me.

Reply:
This is the one tricky part of this change I think.

Basically if a JS function is imported by address only into a wasm file we have to put the JS function into the table.   To do this we need to know its signature.   

The example use case could be a side module that imports `GOT.func.__assert_fail`.   In the old ABI binaryen would rewrite such imports to be `fp$__assert_fail$viiii` and the dynamic linker would extract the signature based on the symbol name.  Binaryen itself was only able to do this because any more that imported `GOT.func.__assert_fail` was assumed to also import `__assert_fail` directly and the signature could be implied from that.

Sadly, even if that later still holds true and each module that imports `GOT.func.__assert_fail` also imports `__assert_fail` we don't have a way with the current JS API to get the expected signature of a function import so its not really feasable to use that trick today.

Also as it happens binaryen can eliminate completely the unused direct import of `__assert_fail`. 

So this is indeed a new limitation:  If you want to reference a JS function my its address in native code you the JS side needs to declare a signature so it can be added to the table.

Technically this is not a new problem.. I'm looking at showing that is kind of exists today if we extent `dlsym` to allow lookup of arbitrary symbols (which I'm doing now on the side).

Reply:
I see, thanks! Makes sense, but please add a summary of that in a comment here.